### ISSUE TITLE
Accessibility needs serious attention in every one of these components

### ISSUE THREAD

**Issue by marcysutton** [buefy/buefy#576]
### Lacking Accessibility in This Component Library

I found this library searching for Vue.js component examples, and sadly it was built with little to no accessibility. This excludes users with disabilities and so many others simply because there is no keyboard or screen reader support.

### Description

This library would work much better for users if accessibility requirements like keyboard operability (reachable, operable, perceivable), focus management, screen reader labeling and text alternatives, programmatically determinable DOM relationships, color contrast, and more. 

Components needing attention:

- Collapse: Associate button with collapsible content and indicate its state. You could try aria-haspopup on the button, aria-controls pointing the button to the content, and aria-expanded to indicate state. 
- Dialog: There are no semantics in use, consider using `role="dialog"` or the `<dialog>` element. Disable background content with `inert` and/or use a focus trap to prevent navigating behind modal dialogs.
- Dropdown: The items inside are not keyboard interactive: anchor tags need `href` attributes to be reachable, or you could use `button` elements instead. Hover menus are completely inaccessible to keyboard users and shouldn't be used at all without a focus counterpart. The menu triggers and items inside need to be programmatically associated like the collapse component above.
- Form Checkbox/Radio: By hiding the native input, you remove the ability for a screen reader to announce that it is an input, and that it is checked/unchecked. Try using a different hiding technique that preserves the inputs for assistive technologies.
- Tabs: There are no tab roles for this component, and the `<a>` elements without `href` means nothing is focusable inside for keyboard users. I'd highly recommend reading up on accessible tabs to understand all of the requirements.

I could easily keep going–nearly every single one of your components was built without a care for accessibility. Quality software is a lot more than about how it looks–please consider the barriers you're putting in place for users of every one of these components.

### Solutions and resources
- Here are some accessible examples of patterns similar to some of yours: http://heydonworks.com/practical_aria_examples/
- Here's the ARIA Authoring Practices Guide, including semantics and navigation conventions for UI components: https://www.w3.org/TR/wai-aria-practices-1.1/
- Here's an article on accessibility in JavaScript apps, including UI component libraries: https://www.smashingmagazine.com/2015/05/client-rendered-accessibility/

### Steps to reproduce

Try using a keyboard or a screen reader to access any of these components.

### Expected behavior

You should be able to interact with all of the components without having to use a mouse or a trackpad. If you turn your screen off, you should still be able to use the components with a screen reader like NVDA or Voiceover.

### Actual behavior

Some real buttons were used, which helps a lot. But there are quite a few components with half-baked or completely forgotten accessibility.

**Comment by chrtz** [comment]
The delete/close buttons, f.g. in the notification, also missing an ARIA-label.

**Comment by DinsmoreDesign** [comment]
I'm really considering making the switch over to Buefy for our Vue apps. Right now, we use Bulma with a custom component library I created, but it'd be awesome to use a library that a larger team is invested in, especially since many of our components are 1:1 copies in functionality to Buefy. The only thing really holding me back right now is accessibility, as this is a huge concern for our team.

It seems like the project is under pretty active development and I certainly don't mind helping address accessibility concerns as a contributor, if it means not having to focus on small bug fixes in the UI all the time, rather than actually writing functional applications :P How quickly do pull requests generally get accepted? The advantage of using our own library right now is, if we find an error, we fix it and push it out and it's done. Heavily using and contributing to an open source project is a little scary for us, if it may affect our sprint timing ;)

**Comment by jtommy** [comment]
@DinsmoreDesign We have just released a new version that contains a new build system and single components but accessibility is an important topic.
Your help could be great!

**Comment by DinsmoreDesign** [comment]
@jtommy  - I've been evaluating a few UI libraries over the past few weeks and I REALLY liked Element... but with its non-responsive nature, that's out. Plus, I like that Buefy subscribes more to the ideology of Bulma, so I don't have to relearn everything. I'm starting a new app now that will be a huge rebuild internally for us and I'm tired of having to deal with updating our own library at this point. I'm going to start porting everything over to Buefy and I'll definitely be adding any accessibility improvements from my library over into Buefy and issue some pull requests!

**Comment by jtommy** [comment]
@DinsmoreDesign Oh great! We're waiting your PRs.

**Comment by jtommy** [comment]
@RiqueBR Sure, we'll improve other components

**Comment by RiqueBR** [comment]
That's awesome! Even if at this point aria-roles were added to the date picker and the select component would be a huge help! Thank you so much for this @jtommy

**Comment by phil-hutchinson** [comment]
I would suggest also adding a role="alert" to snackbars and toasts

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_alert_role

**Comment by service-paradis** [comment]
@phil-hutchinson 
The toast component already have `role="alert"`. Just made a [PR](https://github.com/buefy/buefy/pull/2017) for snackbar.

**Comment by phil-hutchinson** [comment]
@service-paradis 
Oops sorry I guess I only checked the snackbar component. Thanks for taking a look so quickly!

**Comment by ssrcruz** [comment]
@jtommy Are there plans to make the tab labels on the tabs component tab focusable? I know this PR https://github.com/buefy/buefy/pull/647 had a potential fix but it seems it never made it? Just want to know whether I should plan on submitting a PR with this feature.

**Comment by jtommy** [comment]
@ssrcruz yes, please submit your PR :)

**Comment by service-paradis** [comment]
I created a PR with a11y improvements for tabs (#2744). Would love to have some feedback on it!
